libcxx/include/__algorithm/mismatch.h

```diff
@@ -56,6 +56,112 @@ __mismatch(_Iter1 __first1, _Sent1 __last1, _Iter2 __first2, _Pred& __pred, _Pro
 
 #if _LIBCPP_VECTORIZE_ALGORITHMS
 
+template <class _ValueType>
+_LIBCPP_NODISCARD _LIBCPP_HIDE_FROM_ABI _LIBCPP_CONSTEXPR_SINCE_CXX20 __simd_vector<long long, 2>
+__reverse_vector(__simd_vector<long long, 2>& __cmp_res) {
+#  if defined(_LIBCPP_BIG_ENDIAN)
+  __cmp_res = __builtin_shufflevector(__cmp_res, __cmp_res, 1, 0);
+#  endif
+  return __cmp_res;
+}
```

Comment:
This seems very ad-hoc to me. These could all be collapsed into a single
```suggestion
template <class _ValueType, size_t _Np>
_LIBCPP_NODISCARD _LIBCPP_HIDE_FROM_ABI __simd_vector<long long, _Np>
__reverse_vector(__simd_vector<long long, _Np> __cmp_res) {
  return [&]<size_t... _Indices>(index_sequence<_Indices>) {
    return __builtin_shufflevector(__cmp_res, __cmp_res, (_Np - _Indices)...);
  }(make_index_sequence<_Np>{});
}
```
Also note that I removed `_LIBCPP_CONSTEXPR_SINCE_CXX20`, since this is never called in a `constexpr` context. The name of this seems very misleading as well, since this doesn't always reverse the vector.

More generally: Why don't we modify `__find_first_set` to return the first element from left to right on all platforms?

Reply:
Thanks for suggestion, let me know if the current code is fine otherwise, please elaborate on the idea of modifying `__find_first_set` since I'm not sure how the code be common on all platforms.

Reply:
I don't mean that the implementation of `__find_first_set` should be common on all platforms. I mean that it should always return the same information. You probably just have to use `countl_zero` on little endian and `countr_zero` on big endian.